adam-core/src/main/scala/org/bdgenomics/adam/rdd/settheory/SetTheory.scala

```diff
@@ -0,0 +1,346 @@
+package org.bdgenomics.adam.rdd.settheory
+
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.ReferenceRegion
+import scala.collection.mutable.ListBuffer
+
+private[settheory] sealed abstract class SetTheory extends Serializable {
+
+  protected val threshold: Long
+  protected val optPartitionMap: Option[Array[Option[(ReferenceRegion, ReferenceRegion)]]]
+
+  /**
+   * The condition that should be met in order for the primitive to be
```

Comment:
Also, if this package is set theoretic, then we should describe it through set theoretic notation, right? I think this method would be "This method evaluates whether a given pair of regions should be members of the set output by this set theoretic operation." Is that correct?